src/plugins/data/public/search/search_interceptor/utils.ts

```diff
@@ -7,10 +7,8 @@
  */
 
 import stringify from 'json-stable-stringify';
+import { Sha256 } from '../../../../../core/public/utils';
 
 export async function createRequestHash(keys: Record<string, any>) {
-  const msgBuffer = new TextEncoder().encode(stringify(keys));
-  const hashBuffer = await crypto.subtle.digest('SHA-256', msgBuffer);
-  const hashArray = Array.from(new Uint8Array(hashBuffer));
-  return hashArray.map((b) => ('00' + b.toString(16)).slice(-2)).join('');
+  return new Sha256().update(stringify(keys), 'utf8').digest('hex');
```

Comment:
This is the fix to resolve https://github.com/elastic/kibana/issues/97819 (see https://github.com/elastic/kibana/issues/97819#issuecomment-829148557 for the explanation).

After this fix, the functional test started to fail https://github.com/elastic/kibana/pull/98720#issuecomment-829205709 as this change made another bug obvious.  

The change in src/plugins/data/public/search/search_interceptor/search_interceptor.ts fixes that bug and the change in src/plugins/data/public/search/search_interceptor/search_interceptor.test.ts adjusted the unit tests so they would have caught the bug